### PR TITLE
DOC: Add infer frequency to docs

### PR DESCRIPTION
**Pull request by jnmclarty**
Was there a reason infer_freq and inferred_freq weren't in the docs?  Did I pick the right spot to add it? Does the feature work well?

I set out looking for a feature like this, searching "guess frequency", "auto detect frequency".

### CODE REVIEW

File: doc/source/api.rst [hunk 1]
```diff
@@ -1364,6 +1365,7 @@ Time/Date Components
    DatetimeIndex.is_quarter_end
    DatetimeIndex.is_year_start
    DatetimeIndex.is_year_end
+   DatetimeIndex.inferred_freq
```
Comment (jreback) [inline doc/source/api.rst]: this is also defined for ``TimedeltaIndex``

**Comment by jreback** [comment]
minor change, ping when pushed.

**Comment by jorisvandenbossche** [comment]
@jnmclarty Could you also update the `infer_freq` docstring with:

- it also works for TimedeltaIndex
- it raises a ValueError when not having 3 values? (now it only mentions a TypeError when the the provided values are no DatetimeIndex)

**Comment by jreback** [comment]
@jorisvandenbossche side issue, we don't appear to have ``PeriodIndex`` at all in ``api.rst``....do we have an issue for this?

**Comment by jorisvandenbossche** [comment]
Ideally, we should also resolve this one: https://github.com/pydata/pandas/issues/5082

**Comment by jorisvandenbossche** [comment]
Yes, the issue is here: https://github.com/pydata/pandas/issues/6384

**Comment by jreback** [comment]
@jorisvandenbossche on #5082 

In reality, I just think we could just fix ``infer_freq`` & ``infered_freq`` to return an actual offset object (which reprs to a string). rather than adding addtional methods. These compare correctly to strings anyhow.

**Comment by jnmclarty** [comment]
@jreback Okay to merge.

**Comment by jreback** [comment]
thanks!

**Pull request merged**